src/commands/xfr.rs

```diff
@@ -44,9 +50,17 @@ pub struct Xfr {
     #[arg(short = '6', long, conflicts_with = "ipv4")]
     ipv6: bool,
 
-    /// Use only TCP.
+    /// Try UDP first with fallback to TCP, otherwise use only TCP.
+    ///
+    /// Only permitted with IXFR as UDP is not permitted for AXFR.
     #[arg(short, long)]
-    tcp: bool,
```

Comment:
These feel like they should not be boolean flags, but maybe a `--transport=...` flag or something. In general, looking at the CLI after a while, it's hard to spot how some flags interact and what the defaults are. This is something we should look into.